src/main/java/com/example/bookstoreapp/services/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+package com.example.bookstoreapp.services.impl;
+
+import com.example.bookstoreapp.entities.OrderEntity;
+import com.example.bookstoreapp.exceptions.AppRuntimeException;
+import com.example.bookstoreapp.models.Address;
+import com.example.bookstoreapp.models.Cart;
+import com.example.bookstoreapp.models.Order;
+import com.example.bookstoreapp.repositories.OrderEntityRepository;
+import com.example.bookstoreapp.services.AddressService;
+import com.example.bookstoreapp.services.CartItemService;
+import com.example.bookstoreapp.services.OrderService;
+import com.example.bookstoreapp.services.UserContextService;
+import com.example.bookstoreapp.utils.IdGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    @Autowired
+    UserContextService userContextService;
+
+    @Autowired
+    CartItemService cartItemService;
+
+    @Autowired
+    AddressService addressService;
+
+    @Autowired
+    OrderEntityRepository orderEntityRepository;
+    @Autowired
+    EntityManager entityManager;
+
+    @Override
+    public Order createOrder(Order order) {
+        order.setOrderId(IdGenerator.getLongId());
+        order.setCustomerId(userContextService.getUserId());
+        LocalDateTime localDateTime = LocalDateTime.now();
+        order.setOrderTime(localDateTime);
+
+        Cart cart = cartItemService.doCheckout(userContextService.getUserId());
+        order.setTotalAmount(cart.getTotal());
+
+        List<Address> addressList = addressService.findUserAddress(userContextService.getUserId());
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        if (addressList != null) {
+            try {
+                String jsonAddress = objectMapper.writeValueAsString(addressList.get(0));
```

Comment:
What if I have 10 addresses, but here we are considering only the first address?